apis/r/R/SOMADataFrame.R

```diff
@@ -472,26 +472,36 @@ SOMADataFrame <- R6::R6Class(
     #' @param new_domain A named list, keyed by index-column name, with values
     #' being two-element vectors containing the desired lower and upper bounds
     #' for the domain.
-    #' @return No return value
-    tiledbsoma_upgrade_domain = function(new_domain) {
-      # stopifnot("'new_domain' must be CODE ME UP PLZ" = ...
+    #' @param check_only If true, does not apply the operation, but only reports
+    #' whether it would have succeeded.
+    #' @return No return value if `check_only` is `FALSE`. If `check_only` is `TRUE`,
+    #' returns the empty string if no error is detected, else a description of the error.
+    tiledbsoma_upgrade_domain = function(new_domain, check_only = FALSE) {
       # Checking slotwise new shape >= old shape, and <= max_shape, is already
       # done in libtiledbsoma
 
       pyarrow_domain_table <- private$upgrade_or_change_domain_helper(
         new_domain, "tiledbsoma_upgrade_domain"
       )
 
-      invisible(
-        upgrade_or_change_domain(
+      reason_string = (
+         upgrade_or_change_domain(
           self$uri,
           FALSE,
           pyarrow_domain_table$array,
           pyarrow_domain_table$schema,
           .name_of_function(),
+          check_only,
           private$.soma_context
         )
       )
+
+      if (check_only) {
+        return(reason_string)
+      } else {
+        # Return value is always "", or it raises an error trying.
+        invisible(reason_string)
+      }
```

Comment:
There's no check to ensure that `check_only` is `TRUE` or `FALSE`, so checking if `check_only` is `TRUE` prevents issues if someone passes `check_only = NA`
```r
> if (NA) 'yes'
Error in if (NA) "yes" : missing value where TRUE/FALSE needed
```
Also, since the yes-condition of the `if` calls a return, there's no need for the `else` since if the `if` succeeds, the method will exit
```suggestion
      if (isTRUE(check_only)) {
        return(reason_string)
      }

      # Return value is always "", or it raises an error trying.
      return(invisible(reason_string))
```